etc/run-clang-tidy-cached.cc

```diff
@@ -0,0 +1,627 @@
+          break;  // got Ctrl-C
+        }
+        // NOLINTEND
+#endif
+        RepairFilenameOccurences(tmp_out, tmp_out);
+        fs::rename(tmp_out, final_out);  // atomic replacement
+      }
+    };
+
+    std::vector<std::thread> workers;
+    for (auto i = 0; i < kJobs; ++i) {
+      workers.emplace_back(clang_tidy_runner);  // NOLINT
+    }
+    for (auto& t : workers) {
+      t.join();
+    }
+    if (print_progress) {
+      fprintf(stderr, "     \n");  // Clean out progress counter.
+    }
+  }
+
+ private:
+  static fs::path GetCacheBaseDir()
+  {
+    if (const char* from_env = getenv("CACHE_DIR")) {
+      return fs::path{from_env};
+    }
+    if (const char* home = getenv("HOME")) {
+      if (auto cdir = fs::path(home) / ".cache/"; fs::exists(cdir)) {
+        return cdir;
+      }
+    }
+    return fs::path{EnvWithFallback("TMPDIR", "/tmp")};
+  }
+
+  static std::string AssembleArgs(int argc, char** argv)
+  {
+    std::string result = " --quiet";
+    result.append(" '--config-file=").append(GetClangTidyConfig()).append("'");
+    for (const std::string_view arg : kExtraArgs) {
+      result.append(" --extra-arg='").append(arg).append("'");
+    }
+    for (int i = 1; i < argc; ++i) {
+      result.append(" '").append(argv[i]).append("'");
+    }
+    return result;
+  }
+
+  fs::path AssembleProjectCacheDir(const std::string& cache_prefix) const
+  {
+    const fs::path cache_dir = GetCacheBaseDir() / "clang-tidy";
+
+    // Use major version as part of name of our configuration specific dir.
+    auto version = GetCommandOutput(clang_tidy_ + " --version");
+    if (version.empty()) {
+      std::cerr << "Could not invoke " << clang_tidy_ << "; is it in PATH ?\n";
+      exit(EXIT_FAILURE);
+    }
+    std::smatch version_match;
+    const std::string major_version
+        = std::regex_search(
+              version, version_match, std::regex{"version ([0-9]+)"})
+              ? version_match[1].str()
+              : "UNKNOWN";
+
+    // Make sure directory filename depends on .clang-tidy content.
+    hash_t cache_unique_id = hashContent(version + clang_tidy_args_);
+    cache_unique_id ^= hashContent(GetContent(GetClangTidyConfig()));
+    return cache_dir
+           / fs::path(cache_prefix + "v" + major_version + "_"
+                      + ToHex(cache_unique_id, 8));
+  }
+
+  // Fix filename paths found in logfiles that are not emitted relative to
+  // project root in the log - remove that prefix.
+  // (bazel has its own, so if this is bazel, also bazel-specific fix up that).
+  static void RepairFilenameOccurences(const fs::path& infile,
+                                       const fs::path& outfile)
+  {
+    static const std::regex sFixPathsRe = []() {
+      std::string canonicalize_expr = "(^|\\n)(";  // fix names at start of line
+      if (kConfig.is_bazel_project) {
+        auto bzroot = GetCommandOutput("bazel info execution_root 2>/dev/null");
```

Comment:
This will not work in OR

Reply:
`is_bazel_project` is disabled (this is a generic script that can be used for all kinds of situations)